### PR TITLE
ft: S3C-1561 - add quotas to request context

### PR DESCRIPTION
**Pull request by anurag4DSB**
**What does this PR do, and why do we need it?**
Add quotas to request contexts for the quotas feature of Vault

**Which issue does this PR fix?**
S3C-1561

**Special notes for your reviewers:**
This is helper code to check quota on every PUT request of object and part.

### CODE REVIEW

**Review by alexanderchan-scality — APPROVED** [review]

**Review by vrancurel — APPROVED** [review]

**Comment by anurag4DSB** [comment]
@bert-e approve

File: lib/policyEvaluator/RequestContext.js [hunk 1]
```diff
@@ -277,7 +283,7 @@ class RequestContext {
             obj.apiMethod, obj.awsService, obj.locationConstraint,
             obj.requesterInfo, obj.signatureVersion,
             obj.authType, obj.signatureAge, obj.securityToken, obj.policyArn,
-            obj.action);
+            obj.action, obj.quotaDelta);
```
Comment (bennettbuchanan) [inline lib/policyEvaluator/RequestContext.js]: Not following how is this being used?

File: lib/policyEvaluator/RequestContext.js [hunk 2]
```diff
@@ -611,6 +617,10 @@ class RequestContext {
         this._policyArn = policyArn;
         return this;
     }
+
```
Comment (bennettbuchanan) [inline lib/policyEvaluator/RequestContext.js:620]: Can you add JSDoc here?

File: lib/policyEvaluator/RequestContext.js (hunk 1 above)
Comment (anurag4DSB) [reply lib/policyEvaluator/RequestContext.js]: This is bad. I fixed it. Changed it to needQuota.
Great catch, thank you!

File: lib/policyEvaluator/RequestContext.js (hunk 2 above)
Comment (anurag4DSB) [reply lib/policyEvaluator/RequestContext.js:620]: Added  :)

**Comment by anurag4DSB** [comment]
@bert-e reset

**Comment by anurag4DSB** [comment]
@bert-e reset

**Comment by anurag4DSB** [comment]
@bert-e reset

**Comment by anurag4DSB** [comment]
@bert-e reset

**Review by bennettbuchanan — APPROVED** [review]

**Pull request merged**